### ISSUE TITLE
Screenshots are not saved

### ISSUE THREAD

**Issue by hansiemithun** [microsoft/playwright#12774]
Trying to save the screenshots, video on failure.

```
import { PlaywrightTestConfig } from '@playwright/test';

const config: PlaywrightTestConfig = {
  use: {
    trace: 'retain-on-failure',
    screenshot: 'only-on-failure',
    video: 'retain-on-failure',
  },
  reporter: [['list'], ['json', { outputFile: 'testout.json' }]],
};

// eslint-disable-next-line import/no-default-export
export default config;
```

By default, it has to save inside the `test-results` folder. But it is neither created nor saved. I tried to create a `test-results` folder myself if it is related to a permission issue in MacBook, but still no luck, it wipes out the folder every time it runs but after tests fail, I don't see the folder or its related contents captured.

Also  just tried with:

```
use: {
    screenshot: 'on'
  },
```

Still no luck. Can someone assist here if I am missing anything or is this a bug?

**Environment:** Mac OS Big Sur
**Playwright:** 1.20.0
**@playwright/test:** 1.20.0
"electron": "11.0.3"

**Comment by mxschmitt** [comment]
Are you using Electron? Then we can fold it into https://github.com/microsoft/playwright/issues/8208.

**Comment by hansiemithun** [comment]
Yes, we are using electrons. I see a similar ticket: https://github.com/microsoft/playwright/issues/12125 and got closed. All these are addressed in #8208? When can we expect this to be ready?

So at present, tracing, video, and screenshots none work with Electron?

Also, does the playwright have an option of maximizing the window size and test for web and electron?

Thank you

**Comment by mxschmitt** [comment]
> I see a similar ticket: https://github.com/microsoft/playwright/issues/12125 and got closed

Yeah we merged them together since the reason was the same. I'd recommend to upvote the linked one! (Eta = demand = amount of upvotes)

> So at present, tracing, video, and screenshots none work with Electron?

They are working, but you manually need to start tracing, take screenshots or record a video. The wiring between electron and the playwright test test-runner is not yet in place.

> Also, does the playwright have an option of maximizing the window size and test for web and electron?

Not as of today. Fixed viewport makes a test reproducible so we recommend that.
